realm/realm-library/src/objectServer/java/io/realm/SyncManager.java

```diff
@@ -199,6 +218,10 @@ private static synchronized void removeSession(SyncConfiguration syncConfigurati
         if (syncSession != null) {
             syncSession.close();
         }
+        if (sessions.size() == 0) {
```

Comment:
`isEmpty` is doing the same thing http://androidxref.com/7.0.0_r1/xref/libcore/ojluni/src/main/java/java/util/HashMap.java#324 but +1 for clarity